diff --git a/include/configs/wandboard.h b/include/configs/wandboard.h
index 02610df..61cf207 100644
--- a/include/configs/wandboard.h
+++ b/include/configs/wandboard.h
@@ -164,8 +164,8 @@
 		"setenv nextcon 0; " \
 		"if hdmidet; then " \
 			"setenv bootargs ${bootargs} " \
-				"video=mxcfb${nextcon}:dev=hdmi,1280x720M@60," \
-					"if=RGB24; " \
+				"video=mxcfb${nextcon}:dev=hdmi,1920x1080M@60," \
+					"if=RGB24,bpp=32; " \
 			"setenv fbmen fbmem=28M; " \
 			"setexpr nextcon ${nextcon} + 1; " \
 		"else " \
